.github/workflows/maven.yaml

```diff
@@ -0,0 +1,37 @@
+# Adapted from the Github Actions template `ci/maven`
+
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-and-test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
```

Comment:
What is the reason to use JDK 17?

Reply:
The tests fail with Java 8: https://github.com/awslabs/spark-sql-kinesis-connector/actions/runs/14582216670/job/40901147680

I propose that we add this action for Java 11 and 17, and create a task to fix the tests for Java 8 and re-enable the action for Java 8.

Reply:
#55